rest/server_context_test.go

```diff
@@ -971,3 +971,61 @@ func TestDatabaseStartupFailure(t *testing.T) {
 	require.Nil(t, dbContext)
 	require.Equal(t, "Offline", rt.GetDBState())
 }
+
+func TestDatabaseCollectionDeletedErrorState(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Test requires Couchbase Server")
+	}
+	rt := NewRestTesterMultipleCollections(t, nil, 3)
+	defer rt.Close()
+
+	dbConfig := rt.ServerContext().GetDatabaseConfig("db")
+	scopesConfig := GetCollectionsConfig(t, rt.TestBucket, 3)
+	dataStoreNames := GetDataStoreNamesFromScopesConfig(scopesConfig)
+	scope := dataStoreNames[0].ScopeName()
+
+	// remove datastore
+	b, err := base.AsGocbV2Bucket(rt.GetDatabase().Bucket)
+	require.NoError(t, err)
+	dsList, err := b.ListDataStores()
+	require.NoError(t, err)
+	require.NoError(t, b.DropDataStore(dsList[0]))
```

Comment:
I think this might disrupt latter tests that are relying on there being an intact set of collections on this bucket

Either have the test reinstate the datastore we deleted, or create a new datastore just for this test that we're ok with dropping.